test/conformance/walk.go

```diff
@@ -14,415 +14,306 @@ See the License for the specific language governing permissions and
+func getTestName(spec *types.SpecSummary) string {
+	return strings.Join(spec.ComponentTexts[1:], " ")
 }
 
-func (v *visitor) comment(x *ast.BasicLit) string {
-	for _, comm := range v.cMap.Comments() {
-		testOffset := int(x.Pos()-comm.End()) - len("framework.ConformanceIt(\"")
-		//Cannot assume the offset is within three or four tabs from the test block itself.
-		//It is better to trim the newlines, tabs, etc and then we if the comment is followed
-		//by the test block itself so that we can associate the comment with it properly.
-		if 0 <= testOffset && testOffset <= 10 {
-			b1 := make([]byte, x.Pos()-comm.End())
-			//if we fail to open the file to compare the content we just assume the
-			//proximity of the comment and apply it.
-			myf, err := os.Open(v.FileSet.File(x.Pos()).Name())
-			if err == nil {
-				defer myf.Close()
-				if _, err := myf.Seek(int64(comm.End()), 0); err == nil {
-					if _, err := myf.Read(b1); err == nil {
-						if strings.Compare(strings.Trim(string(b1), "\t \r\n"), "framework.ConformanceIt(\"") == 0 {
-							return comm.Text()
-						}
-					}
-				}
-			} else {
-				//comment section's end is noticed within 10 characters from framework.ConformanceIt block
-				return comm.Text()
-			}
-		}
-	}
-	return ""
-}
-
-func (v *visitor) emit(arg ast.Expr) {
-	switch at := arg.(type) {
-	case *ast.BasicLit:
-		if at.Kind != token.STRING {
-			v.failf(at, "framework.ConformanceIt() called with non-string argument")
-			return
-		}
-
-		description := v.getDescription(at.Value)
-		err := validateTestName(description)
+func saveAllTestInfo(dataSet []*conformanceData) {
+	if *confDoc {
+		// Note: this assumes that you're running from the root of the kube src repo
+		templ, err := template.ParseFiles("./test/conformance/cf_header.md")
 		if err != nil {
-			v.failf(at, err.Error())
-			return
+			fmt.Printf("Error reading the Header file information: %s\n\n", err)
 		}
-
-		at.Value = normalizeTestName(at.Value)
-		if *confDoc {
-			v.convertToConformanceData(at)
-		} else {
-			fmt.Printf("%s: %q\n", v.FileSet.Position(at.Pos()).Filename, at.Value)
+		data := struct {
+			Version string
+		}{
+			Version: *version,
 		}
-	default:
-		v.failf(at, "framework.ConformanceIt() called with non-literal argument")
-		fmt.Fprintf(os.Stderr, "ERROR: non-literal argument %v at %v\n", arg, v.FileSet.Position(arg.Pos()))
-	}
-}
+		templ.Execute(os.Stdout, data)
 
-func (v *visitor) getDescription(value string) string {
-	tokens := []string{}
-	for _, describe := range v.describes {
-		tokens = append(tokens, describe.text)
-		if len(describe.lastContext.text) > 0 {
-			tokens = append(tokens, describe.lastContext.text)
+		for _, data := range dataSet {
+			fmt.Printf("## [%s](%s)\n\n", data.TestName, data.URL)
+			fmt.Printf("- Added to conformance in release %s\n", data.Release)
+			fmt.Printf("- Defined in code as: %s\n\n", data.CodeName)
+			fmt.Printf("%s\n\n", data.Description)
```

Comment:
This should have its own bullet, eg
```suggestion
			fmt.Printf("- Test description: %s\n\n", data.Description)
```

Reply:
Do you think it looks better that way? I deliberately didnt put a bullet because in both the old and new version of the docs the description was just a text block (not a bullet). I had added the other fields as a bulleted list to differentiate them from the description.

Reply:
I'm fine with the hard break as well, my engineer brain was going for consistency with the suggestion